core/src/main/java/org/jruby/RubyHash.java

```diff
@@ -1644,6 +1644,24 @@ public IRubyObject select(final ThreadContext context, final Block block) {
         return result;
     }
 
+
+    /** rb_hash_slice
+     *
+     */
+    @JRubyMethod(name = "slice", rest = true)
+    public RubyHash slice(final ThreadContext context, final IRubyObject[] args) {
+        RubyHash result = newHash(getRuntime());
+
+        for (int i = 0; i < args.length; i++) {
+            IRubyObject key = args[i];
+            IRubyObject value = this.internalGet(key);
+            if(value != null)
```

Comment:
if (value != null) { // space after if and we only allow non-curly if it is all on one line.